core/templates/dev/head/pages/collection_editor/editor_tab/CollectionNodeCreatorDirective.js

```diff
@@ -25,18 +25,32 @@ oppia.directive('collectionNodeCreator', [function() {
       'validatorsService', 'CollectionEditorStateService',
       'CollectionLinearizerService', 'CollectionUpdateService',
       'CollectionNodeObjectFactory', 'ExplorationSummaryBackendApiService',
-      'siteAnalyticsService',
+      'siteAnalyticsService', 'UrlInterpolationService',
+      'SearchExplorationsBackendApiService',
       function(
           $scope, $http, $window, $filter, alertsService,
           validatorsService, CollectionEditorStateService,
           CollectionLinearizerService, CollectionUpdateService,
           CollectionNodeObjectFactory, ExplorationSummaryBackendApiService,
-          siteAnalyticsService) {
+          siteAnalyticsService, UrlInterpolationService,
+          SearchExplorationsBackendApiService) {
         $scope.collection = CollectionEditorStateService.getCollection();
         $scope.newExplorationId = '';
         $scope.newExplorationTitle = '';
         var CREATE_NEW_EXPLORATION_URL_TEMPLATE = '/create/<exploration_id>';
 
+        $scope.fetchExplorationsMetadata = function(searchQuery) {
```

Comment:
Hi @kevinlee12  -- I think, showing nothing if no results are found will be better.
Because, typeahead queries on text change and if we will add another directive for the results such that it shows "no results" or maybe if we show "no results" directly -- It will seem a bit buggy as it may work for a specific query at a later stage. What do you suggest ?